Epsilon/Components/CompetenceProfileComponent.cs

```diff
@@ -1,31 +1,308 @@
+                        ){RunProperties = new RunProperties(){FontSize = new FontSize(){Val = "14",},},}
+                    )
+                );
+            
+            headerRow.AppendChild(cell);
+        }
+        
+        table.AppendChild(headerRow);
+        
+        var domainRow = new TableRow();
+            
+
+        domainRow.Append(new TableRowHeight(){Val = 2600,});
+
+        foreach (var row in domain.RowsSet.Types.OrderBy(static c => c.Order))
+        {
+            var cellOutcomes = outcomes.Where(o => o.Row.Id == row.Id).ToList();
+            var types = cellOutcomes.Select(static o => o.Value).ToList();
+            var value = types.MaxBy(static t => t.Order);
+            var count = cellOutcomes.Count.ToString(CultureInfo.InvariantCulture);  
+                
+            var contentCell = new Paragraph(new Run(new Text(count)))
+            {
+                ParagraphProperties = new ParagraphProperties() { Justification = new Justification() { Val = JustificationValues.Center, }, },
+            };
+                
+            var cell = CreateTableCell(
+                "1000",
+                _borderedTableCellBorders,
+                contentCell);
+                
+
+            if (value != null)
+            {
+                cell.TableCellProperties.AppendChild(new Shading { Fill = "#" + value.HexColor, });
+            }
+                
+            domainRow.AppendChild(cell);
+        }
+            
+        table.AppendChild(domainRow);
+        
+        return table;
+    }
+
+    private OpenXmlElement GetTableTwoAxis(LearningDomain domain, List<LearningDomainOutcome> outcomes)
+    {
+        var table = new Table();
+        
+        var tblProp = new TableProperties(
+            new TableWidth(),
+            new TableBorders(
+                new BottomBorder() { Val = new EnumValue<BorderValues>(BorderValues.Single), Size = 3,  },
+                new RightBorder() { Val = new EnumValue<BorderValues>(BorderValues.Single), Size = 10,  },
+                new InsideHorizontalBorder() { Val = new EnumValue<BorderValues>(BorderValues.Single), Size = 3,  },
+                new InsideVerticalBorder() { Val = new EnumValue<BorderValues>(BorderValues.Single), Size = 3,  }
+            )
+        );
+        table.AppendChild<TableProperties>(tblProp);
+        
+        // Define table grid
+        var tblGrid = new TableGrid();
+        table.AppendChild(tblGrid);
+
+        var headerRow = new TableRow();
+
+        var content = new Paragraph(new Run(new Text("")))
+        {
+            ParagraphProperties = new ParagraphProperties() { Justification = new Justification() { Val = JustificationValues.Center, }, },
+        };
+
+        // Empty top-left cell.
+        headerRow.AppendChild(
+            CreateTableCell("700", 
+                _headerTableCellBorders, content));
+        
+        foreach (var col in domain.ColumnsSet.Types.OrderBy(static c => c.Order))
```

Comment:
Se round for each with if statements. A domain is not required to have columns

Reply:
The method itself is in an if statement, domains in the method always have columns